src/components/Pagination.jsx

```diff
@@ -29,6 +34,28 @@ function Pagination({
     onPageChange(currentPage - 1);
   };
 
+  useEffect(() => {
+    // check to see whether to disable left or right arrow buttons on component render
+    if (currentPage >= maxPageNumber) {
+      setDisableRight(true);
+      onPageChange(maxPageNumber);
+    } else {
+      setDisableRight(false);
+    };
+
+    if (currentPage === 1) {
+      setDisableLeft(true);
+    } else {
+      setDisableLeft(false);
+    };
+
+    if (maxPageNumber === 1) {
+      setDisableLeft(true);
+      setDisableRight(true);
+    };
+
+  },[currentPage, onPageSizeOptionChange])
```

Comment:
It is much cleaner and concise to directly use the condition as the value of the "disabled" attribute of the respective buttons instead of maintaining a state and running an effect for this. Since the buttons listen to click events, then running an effect might not be appropriate.